symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/MessageService.java

```diff
@@ -386,4 +406,22 @@ private static Long getEpochMillis(Instant instant) {
   private <T> T executeAndRetry(String name, SupplierWithApiException<T> supplier) {
     return RetryWithRecovery.executeAndRetry(retryBuilder, name, supplier);
   }
+  
+  private File createTemporaryAttachmentFile(Attachment attachment) {
+    if (attachment == null) {
+      return null;
+    }
+    try {
+      String[] fileNameArrays = attachment.filename().split("\\.");
+      if (fileNameArrays.length < 2) {
+        throw new MessageCreationException("Invalid attachment's file name.");
+      }
+      File tempFile = File.createTempFile(fileNameArrays[0], "." + fileNameArrays[fileNameArrays.length - 1]);
```

Comment:
aren't we creating a file using the filename that is already set in the name? (well dropping part of it actually if it has multiple dots)

Reply:
we cannot create a temp file with a specific name. The temp file created by the `File#createTempFile` always has a random string in its name